src/core/public/chrome/chrome_service.test.ts

```diff
@@ -46,350 +47,426 @@ function defaultStartDeps() {
-    "smallLogo": undefined,
-  },
-]
-`);
+        Array [
+          Object {},
+          Object {
+            "logo": "big logo",
+            "smallLogo": "not so big logo",
+          },
+          Object {
+            "logo": "big logo without small logo",
+            "smallLogo": undefined,
+          },
+        ]
+      `);
     });
   });
 
   describe('visibility', () => {
     it('updates/emits the visibility', async () => {
-      const service = new ChromeService({ browserSupportsCsp: true });
-      const start = await service.start(defaultStartDeps());
-      const promise = start
+      const { chrome, service } = await start();
+      const promise = chrome
         .getIsVisible$()
         .pipe(toArray())
         .toPromise();
 
-      start.setIsVisible(true);
-      start.setIsVisible(false);
-      start.setIsVisible(true);
+      chrome.setIsVisible(true);
+      chrome.setIsVisible(false);
+      chrome.setIsVisible(true);
       service.stop();
 
       await expect(promise).resolves.toMatchInlineSnapshot(`
-Array [
-  true,
-  true,
-  false,
-  true,
-]
-`);
+        Array [
+          true,
+          true,
+          false,
+          true,
+        ]
+      `);
     });
 
     it('always emits false if embed query string is in hash when set up', async () => {
       window.history.pushState(undefined, '', '#/home?a=b&embed=true');
 
-      const service = new ChromeService({ browserSupportsCsp: true });
-      const start = await service.start(defaultStartDeps());
-      const promise = start
+      const { chrome, service } = await start();
+      const promise = chrome
+        .getIsVisible$()
+        .pipe(toArray())
+        .toPromise();
+
+      chrome.setIsVisible(true);
+      chrome.setIsVisible(false);
+      chrome.setIsVisible(true);
+      service.stop();
+
+      await expect(promise).resolves.toMatchInlineSnapshot(`
+        Array [
+          false,
+          false,
+          false,
+          false,
+        ]
+      `);
+    });
+
+    it('application-specified visibility on mount', async () => {
+      const startDeps = defaultStartDeps();
+
+      // We can fake the application service navigating to the next app by
+      // adding a fake app to the application's available apps then triggering
+      // the next ID.
+      startDeps.application.availableApps = new Map([
+        createApp('alpha'),
+        createApp('beta', true),
+        createApp('gamma', false),
+      ]);
+
+      const { chrome, service } = await start({ startDeps });
+      const promise = chrome
+        .getIsVisible$()
+        .pipe(toArray())
+        .toPromise();
+
+      startDeps.application.currentAppId$.next('alpha');
+      startDeps.application.currentAppId$.next('beta');
+      startDeps.application.currentAppId$.next('gamma');
+      service.stop();
+
+      await expect(promise).resolves.toMatchInlineSnapshot(`
+        Array [
+          true,
+          false,
+          true,
+          false,
```

Comment:
This looks backwards as well. Just looking at the last case, the `gamma` app has `chromeHidden: false` but the last visibility is `false`? Shouldn't it be true?